aztk/utils/ssh.py

```diff
@@ -0,0 +1,98 @@
+'''
+    SSH utils
+'''
+import asyncio
+import io
+import os
+import select
+import socketserver as SocketServer
+import sys
+from concurrent.futures import ThreadPoolExecutor
+
+import paramiko
+
+from . import helpers
+
+
+def connect(hostname,
+            port=22,
+            username=None,
+            password=None,
+            pkey=None):
+
+    client = paramiko.SSHClient()
+
+    client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+
+    if pkey:
+        ssh_key = paramiko.RSAKey.from_private_key(file_obj=io.StringIO(pkey))
+    else:
+        ssh_key = None
+
+    client.connect(
+        hostname,
+        port=port,
+        username=username,
+        password=password,
+        pkey=ssh_key
+    )
+
+    return client
+
+
+def node_exec_command(command, container_name, username, hostname, port, ssh_key=None, password=None):
+    client = connect(hostname=hostname, port=port, username=username, password=password, pkey=ssh_key)
+    docker_exec = 'sudo docker exec 2>&1 -t {0} /bin/bash -c \'set -e; set -o pipefail; {1}; wait\''.format(container_name, command)
+    stdin, stdout, stderr = client.exec_command(docker_exec, get_pty=True)
+    [print(line.decode('utf-8')) for line in stdout.read().splitlines()]
```

Comment:
should you also print stderr?

Reply:
in line 45, I'm redirecting stdout and stderr so they are returned together